lib/pc_patch.c

```diff
@@ -522,6 +522,77 @@ pc_patch_from_patchlist(PCPATCH **palist, int numpatches)
 	return (PCPATCH*)paout;
 }
 
+// first: the first element to select.
+// count: the number of element to select from the first one.
+PCPATCH *
+pc_patch_range(const PCPATCH *pa, int first, int count)
+{
+	PCPATCH_UNCOMPRESSED *paout;
+	uint8_t *buf;
+	size_t size;
+	size_t start;
+
+	assert(pa);
+
+	if(first+count > pa->npoints)
+	count = pa->npoints - first;
+
+	paout = pc_patch_uncompressed_make(pa->schema, count);
+	buf = paout->data;
+	paout->npoints = count;
+	start = pa->schema->size * first;
+	size = pa->schema->size * count;
+
+	switch( pa->type )
```

Comment:
This switch is somewhat redundant with pc_patch_uncompress, which could be used instead (freeing the uncompressed patch if its pointer differs from the input patch.)